Branch-SDK/Branch-SDK/BNCLinkData.m

```diff
@@ -124,12 +126,13 @@ - (NSUInteger)hash {
     NSUInteger result = 1;
     NSUInteger prime = 19;
 
+    NSString *encodedParams = [BNCEncodingUtils encodeDictionaryToJsonString:self.params];
     result = prime * result + self.type;
     result = prime * result + [[BNCEncodingUtils md5Encode:[self.alias lowercaseString]] hash];
     result = prime * result + [[BNCEncodingUtils md5Encode:[self.channel lowercaseString]] hash];
     result = prime * result + [[BNCEncodingUtils md5Encode:[self.feature lowercaseString]] hash];
     result = prime * result + [[BNCEncodingUtils md5Encode:[self.stage lowercaseString]] hash];
-    result = prime * result + [[BNCEncodingUtils md5Encode:[self.params lowercaseString]] hash];
+    result = prime * result + [[BNCEncodingUtils md5Encode:[encodedParams lowercaseString]] hash];
```

Comment:
regardless the others above, but we can get rid of lowercaseString for this line for sure.